src/components/Header.tsx

```diff
@@ -30,96 +43,38 @@ interface HeaderProps {
   onClickPrevious: () => void;
 }
 
-const MONTHS = ["Jan", "Feb", "Mar", "Apr", "May", "June", "July", "Aug", "Sept", "Oct", "Nov", "Dec"];
-
-const generateYears = (relativeTo: Date, count: number) => {
-  const half = Math.floor(count / 2);
-  return Array(count)
-    .fill(0)
-    .map((_y, i) => relativeTo.getFullYear() - half + i); // TODO: make part of the state
-};
-
 const Header: React.FunctionComponent<HeaderProps> = ({
   date,
-  setDate,
   nextDisabled,
   prevDisabled,
   onClickNext,
   onClickPrevious,
 }: HeaderProps) => {
   const classes = useStyles();
 
-  const handleMonthChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
-    setDate(setMonth(date, parseInt(event.target.value)));
-  };
-
-  const handleYearChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
-    setDate(setYear(date, parseInt(event.target.value)));
-  };
+  const { getMonth, getYear } = useDateRangeContext();
 
   return (
     <Grid container justify="space-between" alignItems="center">
-      <Grid item className={classes.iconContainer}>
-        <IconButton className={classes.icon} disabled={prevDisabled} onClick={onClickPrevious}>
-          <ChevronLeft color={prevDisabled ? "disabled" : "action"} />
-        </IconButton>
-      </Grid>
-      <Grid item>
-        <Select
-          value={getMonth(date)}
-          onChange={handleMonthChange}
-          MenuProps={{
-            anchorOrigin: {
-              vertical: "bottom",
-              horizontal: "left",
-            },
-            transformOrigin: {
-              vertical: "top",
-              horizontal: "left",
-            },
-            getContentAnchorEl: null,
-          }}
-          className={classes.header}
-        >
-          {MONTHS.map((month, idx) => (
-            <MenuItem key={month} value={idx} className={classes.header}>
-              {month}
-            </MenuItem>
-          ))}
-        </Select>
-      </Grid>
+      {prevDisabled || (
+        <Grid item className={classes.iconContainer}>
+          <IconButton className={classes.icon} onClick={onClickPrevious}>
+            <ChevronLeft color={prevDisabled ? "disabled" : "action"} />
```

Comment:
If `prevDisabled` is true, the component will never be rendered. We can safely use "action" only.